src/games/gameitem.py

```diff
@@ -214,103 +214,31 @@ def tooltip(self, data):
-                    yield 'middle', team
-
-        team_tables = [self._team_table(team, align)
-                       for align, team in alignment(teams)]
-        return versus_string.join(team_tables)
-
-    def _team_table(self, team, align):
-        team_table_start = "<td><table>"
-        team_table_end = "</table></td>"
-        rows = [self._player_table_row(player, align) for player in team]
-        return team_table_start + "".join(rows) + team_table_end
-
-    def _player_table_row(self, player, align):
-        if isinstance(player, str):
-            country = "<td></td>"
-        else:
-            country = "<td>{country_icon}</td>"
-        pname = ("<td align='{alignment}' valign='middle' width='135'>"
-                 "{player}"
-                 "</td>")
-        order = [pname, country] if align == 'right' else [country, pname]
-        player_row = "<tr>{}{}</tr>".format(*order)
-
-        if isinstance(player, str):
-            return player_row.format(alignment=align, player=player)
-        else:
-            return player_row.format(
-                country_icon=self._country_icon_fmt(player),
-                alignment=align,
-                player=self._player_fmt(player))
-
-    def _country_icon_fmt(self, player):
-        icon_path_fmt = os.path.join("chat", "countries", "{}.png")
-        icon_path = icon_path_fmt.format(player.country.lower())
-        icon_abs_path = os.path.join(util.COMMON_DIR, icon_path)
-        return "<img src='{}'>".format(icon_abs_path)
-
-    def _player_fmt(self, player):
-        if player == self._me.player:
-            pformat = "<b><i>{}</b></i>"
-        else:
-            pformat = "{}"
-        player_string = pformat.format(player.login)
-        if player.rating_deviation < 200:   # FIXME: magic number
-            player_string += " ({})".format(player.rating_estimate())
-        return player_string
-
-    def _observers_tooltip(self, observers):
-        if not observers:
-            return ""
-
-        observer_fmt = "{country_icon} {observer}"
-
-        observer_strings = [observer_fmt.format(
-            country_icon=self._country_icon_fmt(observer),
-            observer=observer.login)
-            for observer in observers]
-        return "Observers: " + ", ".join(observer_strings)
-
-    def _mods_tooltip(self, mods):
-        if not mods:
-            return ""
-        return "<br/>With: " + "<br/>".join(mods.values())
 
     def format(self, teams, observers, mods):
-        teamtip = self._teams_tooltip(teams)
-        obstip = self._observers_tooltip(observers)
-        modtip = self._mods_tooltip(mods)
-
-        if modtip:
-            modtip = "<br/>" + modtip
-
-        return self.TIP_FORMAT.format(teams=teamtip,
-                                      observers=obstip,
-                                      mods=modtip)
+        def get_file_path(player):
+            if player is None or isinstance(player, str):
+                return
```

Comment:
Why do we need that? How can be possible, that player is `None` or `str`? O_o

`Funcitons should do one thing, and should ti it well.` There you have logic, that check proper data, i think, you should send here clear data.

Reply:
a player can be none when the tooltip is opened before the client fetches the data as far as i know - not 100% sure, this might be obsolete.
if a player is a string then he appears offline to the client (same server issue i mentioned above)